iam-login-service/src/main/java/it/infn/mw/iam/core/oauth/profile/common/BaseAccessTokenBuilder.java

```diff
@@ -152,6 +164,14 @@ protected JWTClaimsSet.Builder baseJWTSetup(OAuth2AccessTokenEntity token,
       audience = authentication.getOAuth2Request().getRequestParameters().get(AUD_KEY);
     }
 
+    if (hasCustomValidityRequest(authentication)) {
+      expiry = authentication.getOAuth2Request().getRequestParameters().get(EXPIRES_IN_KEY);
+    }
+
+    if(hasCustomValidityRefreshFlow(authentication)){
```

Comment:
If I understood well how this framework works, probably you can check this condition (and understand if it's a refresh-flow) or else check the previous one. Doing this, it's unnecessary to filter the `expires_in` parameter inside IamTokenService.

```
if (hasCustomValidityRefreshFlow(authentication)) {
  expiry = authentication.getOAuth2Request().getRequestParameters().get(EXPIRES_IN_KEY);
} else if (hasCustomValidityRequest(authentication)) {
  expiry = authentication.getOAuth2Request().getRequestParameters().get(EXPIRES_IN_KEY);
}